### PR TITLE
Refactor code base to use async-await more and other es6+ features

### PR DESCRIPTION
**Pull request by JakeChampion**
Biggest performance win in this is replacing a set of nested loops with a single `.includes` call.

Hopefully this refactoring helps make the code more maintainable as well.

resolves https://github.com/Financial-Times/polyfill-library/issues/177

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -2,41 +2,41 @@
...
 
 module.exports = {
-	listAliases,
-	listAllPolyfills,
-	describePolyfill,
-	getOptions,
-	getPolyfills,
-	getPolyfillString,
-	normalizeUserAgent
+  listAliases,
```
Comment (chee) [inline lib/index.js]: switching from tabs to spaces in some files?

File: lib/index.js [hunk 2]
```diff
@@ -60,27 +60,33 @@ function describePolyfill(featureName) {
...
-	};
+  const hasProperty = (prop, obj) =>
+    Object.prototype.hasOwnProperty.call(obj, prop);
+  const options = {
+    callback:
+      hasProperty("callback", opts) &&
+      typeof opts.callback === "string" &&
+      opts.callback.match(/^[\w\.]+$/)
+        ? opts.callback
+        : false,
```
Comment (chee) [inline lib/index.js]: standard es6 syntax

File: lib/index.js [hunk 3]
```diff
@@ -99,100 +105,98 @@ function getOptions(opts = {}) {
  * @return {Promise<Object>} - Canonicalised feature definitions filtered for UA
  */
 function getPolyfills(opts) {
```
Comment (chee) [inline lib/index.js]: this is nicer

File: lib/index.js (hunk 1 above)
Comment (JakeChampion) [reply lib/index.js]: oops! I'll fix that!

**Pull request merged**